src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -1007,15 +1025,26 @@ private void SetOutputPorts()
             // Clear out all the output port models
             OutPorts.RemoveAll((p) => true);
 
+            int i = 0;
+
             foreach (var def in allDefs)
             {
+                if (i >= outputPortNames.Count) break;
```

Comment:
This is a finite loop. It runs for each `def` in `allDefs`. Why are you breaking out of it in between?